go-controller/pkg/ovn/service.go

```diff
@@ -285,11 +286,34 @@ func (ovn *Controller) createService(service *kapi.Service) error {
 }
 
 func (ovn *Controller) updateService(oldSvc, newSvc *kapi.Service) error {
-	klog.V(5).Infof("Updating service is a noop: %s", newSvc.Name)
-	// Service update needs to check for port change, protocol, etc and update the ACLs, or and OVN LBs
-	// This only really matters when a service is updated and has no endpoints. If the service has endpoints
-	// the endpoints watcher will handle the OVN config
-	// TODO (trozet) implement this
+	if reflect.DeepEqual(newSvc.Spec.Ports, oldSvc.Spec.Ports) && reflect.DeepEqual(newSvc.Spec.ExternalIPs, oldSvc.Spec.ExternalIPs) {
+		klog.V(5).Infof("skipping service update as change does not apply to .Spec.Ports or .Spec.ExternalIP")
+		return nil
+	}
+
+	if !util.IsClusterIPSet(newSvc) && !util.IsClusterIPSet(oldSvc) {
+		klog.V(5).Infof("skipping service update due to service being headless")
+		return nil
+	}
+
+	klog.V(5).Infof("updating service from: %v to: %v", oldSvc, newSvc)
+
+	// New service or old service, it's all the same for what concerns the endpoint.
+	ep, err := ovn.watchFactory.GetEndpoint(newSvc.Namespace, newSvc.Name)
+	if err != nil {
+		klog.V(5).Infof("updating service with no endpoint, service: %s, namespace: %s", newSvc.Name, newSvc.Namespace)
+	}
+
+	ovn.deleteService(oldSvc)
+
+	if err := ovn.createService(newSvc); err != nil {
+		return err
+	}
+	if ep != nil {
+		if err := ovn.AddEndpoints(ep); err != nil {
```

Comment:
if there is a service update, does it also trigger an endpoint update event? Depending on the answer this code needs to change.

Reply:
We don't need to do AddEndpoints here since we createService again (which will call it). I know we already agreed on this offline, but just updating it for others who read the comment.